keps/sig-instrumentation/2831-kubelet-tracing/README.md

```diff
@@ -0,0 +1,329 @@
+
+OpenTelemetry-Go provides the [propagation package](https://github.com/open-telemetry/opentelemetry-go/blob/main/propagation/propagation.go) with which you can add custom key-value pairs known as [baggage](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/baggage/api.md). Baggage data will be propagated across services within contexts.
+
+### Running the OpenTelemetry Collector
+
+The [OpenTelemetry Collector](https://github.com/open-telemetry/opentelemetry-collector) can be run as a sidecar, a daemonset, a deployment , or a combination in which the daemonset buffers telemetry and forwards to the deployment for aggregation (e.g. tail-base sampling) and routing to a telemetry backend.  To support these various setups, the kubelet should be able to send traffic either to a local (on the control plane network) collector, or to a cluster service (on the cluster network).
+
+### Kubelet Configuration
+
+```golang
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// KubeletConfiguration contains the configuration for the Kubelet
+type KubeletConfiguration struct {
+    metav1.TypeMeta
+    ------------
+    // EnableOtelTracing enables export of opentelemetry traces
+    // +optional
+    EnableOtelTracing bool
+    // OpenTelemetryConfig
+    // +optional
+    OpenTelemetryConfig OpenTelemetryConfig
+}
+
+// OpenTelemetryConfig specifies configuration for opentelemetry tracing
+type OpenTelemetryConfig struct {
+    // +optional
+    // The name of the tracing service
+    // Defaults to node hostname. This results in a tracing service for each node.
+    TracingServiceName string
+
+    // +optional
+    // The port of the collector agent running on the node
+    // Defaults to 0.0.0.0:4317
+    CollectorEndpoint  string
+
+    // +optional
+    // SamplingRatePerMillion is the number of samples to collect per million spans.
+    // Defaults to 0.
+    SamplingRatePerMillion *int32
+}
+```
+
+### Test Plan
+
+We will test tracing added by this feature with an integration test.  The
+integration test will verify that spans exported by the kubelet match what is
```

Comment:
Should probably test the connection of the trace between the kubelet and apiserver as well.

Reply:
added that